cpp/examples/history_example.cpp

```diff
@@ -0,0 +1,33 @@
+#include "memilio/io/history.h"
+#include <iostream>
+#include <vector>
+
+// Define a Logger
+struct MyLogger : mio::LogAlways {
+    using Type = int;
+    static Type log(const int& input)
+    {
+        return input * 2;
+    }
+};
+
+int main()
+{
+    // Create a History object
+    mio::History<mio::DataWriterToMemory, MyLogger> history;
```

Comment:
Please explain a little more here, this is an introductory example. E.g. explain the choice of Writer and where it comes from, mention that multiple loggers can be used etc.

Reply:
yes